File: python/_modulos/pilha.py
class Pilha:
    def __init__(self):
        self.max = 100
        self.pilha = [None]*self.max
        self.topo = -1

    def empilhar(self, valor):
        self.topo += 1

        if self.topo == self.max:
            self.max *= 2
            temp = [None]*self.max

            for i in range(int(self.max / 2)):
                temp[i] = self.pilha[i]

            self.pilha = temp

            self.pilha[self.topo] = valor
        else:
            self.pilha[self.topo] = valor

    def desempilhar(self):
        if self.topo == -1:
            print("Pilha vazia")
        else:
            elemento = self.pilha[self.topo]
            self.topo -= 1
            return elemento
        
    def imprimir(self):
        for i in range(self.topo + 1):
            print(self.pilha[i], end=" ")

    def tamanho(self):
        return self.topo + 1

File: javascript/19_pilha/pilha.js
const pilha = (function() {
	var max = 100;
	var pilha = new Array(max);
	var topo = -1;
	
	function empilhar(valor) {
		topo++
		
		if (topo == max) {
			max *= 2;
			var temp = new Array(max);
			
			for (var i = 0; i < max / 2; i++) {
				temp[i] = pilha[i];
			}
			
			pilha = temp;
			
			pilha[topo] = valor;
		} else {
			pilha[topo] = valor;
		}
	}
	
	function desempilhar() {
		if (topo == -1) {
			document.write("Pilha vazia<br>")
		} else {
			var elemento = pilha[topo];
			topo--;
			return elemento;
		}
	}
	
	function imprimir() {
		for (var i = 0; i <= topo; i++) {
			document.write(pilha[i] + " ");
		}
	}
	
	function tamanho() {
		return topo + 1;
	}
	
	return {
		empilhar: empilhar,
		desempilhar: desempilhar,
		imprimir: imprimir,
		tamanho: tamanho
	}
}());